plugins/destination/postgresql/client/migrate.go

```diff
@@ -3,210 +3,326 @@ package client
-		sort.Strings(sortedPKs)
-		for _, pk := range sortedPKs {
-			stalePk := true
-			for _, col := range table.Columns {
-				if col.Name == pk && col.CreationOptions.PrimaryKey {
-					stalePk = false
-					break
-				}
-			}
-			if stalePk {
-				c.logger.Info().Str("table", table.Name).Str("column", pk).Msg("Column exists with primary key but is not in the schema")
-				stalePks = append(stalePks, pk)
+func (c *Client) normalizeTablePg(table *schema.Table) *schema.Table {
+	for i := range table.Columns {
+		if !c.enabledPks() {
+			table.Columns[i].CreationOptions.PrimaryKey = false
+		}
+		if table.Columns[i].CreationOptions.PrimaryKey {
+			table.Columns[i].CreationOptions.NotNull = true
+		}
+		// this is for backward compatibility as I believe this is as of right now defined on the source
+		if c.enabledPks() && len(table.PrimaryKeys()) == 0 {
+			cqIdColumn := table.Columns.Get(schema.CqIDColumn.Name)
+			if cqIdColumn != nil {
+				cqIdColumn.CreationOptions.PrimaryKey = true
 			}
 		}
 	}
-	return stalePks
+	return table
 }
 
-func (c *Client) autoMigrateTable(ctx context.Context, table *schema.Table) error {
-	var err error
-	var pgColumns *pgTableColumns
-	var pgPKs map[string]bool
-
-	// create the new column as it doesn't exist
-	tableName := pgx.Identifier{table.Name}.Sanitize()
-	if pgColumns, err = c.getPgTableColumns(ctx, table.Name); err != nil {
-		return fmt.Errorf("failed to get table %s columns types: %w", table.Name, err)
-	}
-
-	if pgPKs, err = c.getPgTablePrimaryKeys(ctx, table.Name); err != nil {
-		return fmt.Errorf("failed to get table %s primary key columns: %w", table.Name, err)
+func (c *Client) normalizeTable(table *schema.Table) *schema.Table {
+	switch c.pgType {
+	case pgTypeCockroachDB:
+		return c.normalizeTableCockroach(table)
+	case pgTypePostgreSQL:
+		return c.normalizeTablePg(table)
+	default:
+		panic("unknown pg type")
 	}
+}
 
-	stalePks := c.getStalePks(pgPKs, table)
-	constraintName := pgx.Identifier{table.Name + "_cqpk"}.Sanitize()
-	dropConstraintSQL := "alter table " + tableName + " drop constraint if exists " + constraintName
-	if len(stalePks) > 0 {
-		message := "the following primary keys were removed from the schema %q for table %q.\nYou can migrate the table manually by running:\n%s"
-		sep := strings.Repeat("-", len(dropConstraintSQL)+1)
-		query := fmt.Sprintf("%s\n%s;\n%s\n%s", sep, dropConstraintSQL, getDropNotNullQuery(table, stalePks), sep)
-		return fmt.Errorf(message, stalePks, table.Name, query)
-	}
-
-	reCreatePrimaryKeys := false
-
-	for _, col := range table.Columns {
-		columnName := pgx.Identifier{col.Name}.Sanitize()
-		columnType := c.SchemaTypeToPg(col.Type)
-		if columnType == "" {
-			c.logger.Warn().Str("table", table.Name).Str("column", col.Name).Msg("Column type not supported, skipping")
+func (c *Client) autoMigrateTable(ctx context.Context, table *schema.Table, changes []schema.TableColumnChange) error {
+	for _, change := range changes {
+		switch change.Type {
```

Comment:
Can we simplify this to:
```
if change.Type == schema.TableColumnChangeTypeAdd {
  if err := c.addColumn(ctx, table.Name, change.Current); err != nil {
    return err
  }
}
```

and drop the other cases? We already know the table is auto migratable when we call this method

Reply:
Updated.

Reply:
It was more of a safe way to know if we missed anything.